packages/server/graphql/mutations/archiveTimelineEvent.ts

```diff
@@ -0,0 +1,71 @@
+import {GraphQLID, GraphQLNonNull} from 'graphql'
+import getRethink from '../../database/rethinkDriver'
+import {GQLContext} from '../graphql'
+import ArchiveTimelineEventPayload from '../types/ArchiveTimelineEventPayload'
+import publish from '../../utils/publish'
+import {SubscriptionChannel} from 'parabol-client/types/constEnums'
+import TimelineEvent from '../../database/types/TimelineEvent'
+import {getUserId, isTeamMember} from '../../utils/authorization'
+import standardError from '../../utils/standardError'
+
+const archiveTimelineEvent = {
+  type: GraphQLNonNull(ArchiveTimelineEventPayload),
+  description: `Archive a timeline event`,
+  args: {
+    timelineEventId: {
+      type: GraphQLNonNull(GraphQLID),
+      description: 'the id for the timeline event'
+    },
+    meetingId: {
+      type: GraphQLNonNull(GraphQLID),
+      description: 'the meeting id for the timeline event'
+    },
+    teamId: {
+      type: GraphQLNonNull(GraphQLID),
+      description: 'the team id for the timeline event'
+    }
+  },
+  resolve: async (
+    _source,
+    {timelineEventId, meetingId, teamId},
+    {authToken, dataLoader, socketId: mutatorId}: GQLContext
+  ) => {
+    const r = await getRethink()
+    const operationId = dataLoader.share()
+    const subOptions = {mutatorId, operationId}
+    const viewerId = getUserId(authToken)
+
+    // AUTH
+    if (!isTeamMember(authToken, teamId)) {
```

Comment:
this is still vulnerable to an attacker.
Imagine I'm an attacker who knows your `meetingId`.
There's nothing preventing me from calling this mutation with my `teamId` & your `meetingId`.

you could fix this by doing the following:

```
const meeting = await dataLoader.get('meetings').load(meetingId)
if (!meeting) {
  return {error: {message: 'Meeting not found'}}
}
const {teamId} = meeting
if (isTeamMember(teamId, authToken)) {
  return {error: {message: 'Meeting not found'}}
}
```

By determining the `teamId` on the server we don't have to trust the client.